packages/components/src/input-control/reducer/reducer.ts

```diff
@@ -206,31 +199,33 @@ export function useInputControlStateReducer(
 
 	const currentState = useRef( state );
 	const currentValueProp = useRef( initialState.value );
+	const refEvent = useRef< SyntheticEvent | null >( null );
 	useLayoutEffect( () => {
 		currentState.current = state;
 		currentValueProp.current = initialState.value;
 	} );
 	useLayoutEffect( () => {
+		if ( ! refEvent.current ) return;
+
 		if (
 			state.value !== currentValueProp.current &&
 			! currentState.current.isDirty
 		) {
 			onChangeHandler( state.value ?? '', {
-				event: currentState.current._event as
+				event: refEvent.current as
 					| ChangeEvent< HTMLInputElement >
 					| PointerEvent< HTMLInputElement >,
 			} );
 		}
-	}, [ state.value ] );
+		refEvent.current = null;
+	}, [ state.value, state.isDragging, isFocused ] );
```

Comment:
The additional dependencies were added only in the interest of executing `refEvent.current = null` not `onChangeHandler`. That's done as part of making sure the effect hook used to intake values doesn't dispatch (via `reset` in the base PR) more than necessary.

Without using the existence of the event to avoid some dispatches, there was [an issue](https://github.com/WordPress/gutenberg/runs/6166230703?check_suite_focus=true#step:5:920) with `FocalPointPicker`. That component is surely doing it wrong but it seemed the shorter and more defensive path to reduce the dispatches.

I'm glad to take another look and see how it might be improved upon. I have some ideas.

Reply:
I've pushed 2b14b0f, which removes the additional dependencies. The event ref (if it exists) is now cleared when the intake effect hook runs. This is much better. Thank you for the critique @youknowriad 🙇